### ISSUE TITLE
Environment breaking ItemGroups added when I add, rename, or copy and paste files in solution explorer

### ISSUE THREAD

**Issue by jons-aura** [dotnet/project-system#7028]
**Visual Studio Version**:

16.8.4

**Summary**:

Same issue as here I think. https://github.com/dotnet/project-system/issues/2810
I'm getting environment breaking ItemGroups added when I add, rename, or copy and paste files in solution explorer.

**Steps to Reproduce**:

1. Rename a file in VS

2. csproj gets updated

3. VS freaks out and I have to remove the spurious csproj entries and *entirely close and reopen VS*

**Expected Behavior**:

csproj should not be updated.

**Actual Behavior**:

**User Impact**:

**Comment by jjmew** [comment]
@jons-aura Is this on an SDK style project? can you send us the before and after csproj file?

**Comment by jons-aura** [comment]
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <Nullable>warnings</Nullable>
    <TargetFramework>netstandard2.1</TargetFramework>
    <DebugType>Full</DebugType>
    <!--<Nullable>enable</Nullable>-->
    <!--<WarningsAsErrors>nullable</WarningsAsErrors>-->
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <WarningsAsErrors>;NU1605;CS0108</WarningsAsErrors>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="CsvHelper" Version="16.0.0" />
    <PackageReference Include="Hangfire.Core" Version="1.7.17" />
    <PackageReference Include="IronPdf" Version="2020.10.3.3" />
    <PackageReference Include="Microsoft.AspNetCore.Cryptography.KeyDerivation" Version="5.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.OData" Version="7.5.2" />
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Options" Version="5.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
    <PackageReference Include="Sendgrid" Version="9.21.1" />
    <PackageReference Include="Serilog" Version="2.10.0" />
    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
    <PackageReference Include="System.DirectoryServices" Version="5.0.0" />
    <PackageReference Include="System.DirectoryServices.AccountManagement" Version="5.0.0" />
  </ItemGroup>
  
  <ItemGroup>
    <ProjectReference Include="XXXX" />
  </ItemGroup>

</Project>
```
becomes 
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <Nullable>warnings</Nullable>
    <TargetFramework>netstandard2.1</TargetFramework>
    <DebugType>Full</DebugType>
    <!--<Nullable>enable</Nullable>-->
    <!--<WarningsAsErrors>nullable</WarningsAsErrors>-->
  </PropertyGroup>

  <PropertyGroup Condition="'$(Configuration)|$(Platform)'=='Debug|AnyCPU'">
    <WarningsAsErrors>;NU1605;CS0108</WarningsAsErrors>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Models\WebSpecialAssignmentDto - Copy.cs" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="CsvHelper" Version="16.0.0" />
    <PackageReference Include="Hangfire.Core" Version="1.7.17" />
    <PackageReference Include="IronPdf" Version="2020.10.3.3" />
    <PackageReference Include="Microsoft.AspNetCore.Cryptography.KeyDerivation" Version="5.0.0" />
    <PackageReference Include="Microsoft.AspNetCore.OData" Version="7.5.2" />
    <PackageReference Include="Microsoft.CodeAnalysis.FxCopAnalyzers" Version="3.3.1">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
    <PackageReference Include="Microsoft.Extensions.Options" Version="5.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="12.0.3" />
    <PackageReference Include="Sendgrid" Version="9.21.1" />
    <PackageReference Include="Serilog" Version="2.10.0" />
    <PackageReference Include="System.Data.SqlClient" Version="4.8.2" />
    <PackageReference Include="System.DirectoryServices" Version="5.0.0" />
    <PackageReference Include="System.DirectoryServices.AccountManagement" Version="5.0.0" />
  </ItemGroup>
  
  <ItemGroup>
    <ProjectReference Include="XXXX" />
  </ItemGroup>

</Project>

**Comment by drewnoakes** [comment]
@jons-aura I agree that VS probably shouldn't be adding the `Compile` item to your project, but I cannot see how that would cause any actual problems.

> VS freaks out and I have to remove the spurious csproj entries and entirely close and reopen VS

Can you be more specific about the problem so that we can further diagnose?

**Comment by jons-aura** [comment]
I couldn't get it to "freak out" by copy+pasting but when I renamed a file in Solution Explorer it did.
![image](https://user-images.githubusercontent.com/35536253/111628576-be3ee780-87c6-11eb-938a-42ae1860fb28.png)

As you can see the renamed file shows up as added and fine but every other file in the project now shows as not being part of the project. If I right click -> unload and then right click -> reload it doesn't help. If I edit the .csproj and remove the new "Compile" item and unload, reload that doesn't help either, I have to close and reopen VS entirely to get it to load correctly again.

**Comment by drewnoakes** [comment]
@jons-aura if you're able to end up with a missing item (item with a red cross) by using copy/paste and rename in VS, then there is a bug. I've tried this out locally and couldn't get a repro. To understand what's happening we need a step-y-step breakdown, with some more screenshots at various points. I'm particularly concerned that there is a state for which restarting VS made a difference, but reloading the project did not. Can you provide steps to get into that state please? It would be ideal if you started with a blank project, so that we can rule out other customisations to the project.

**Comment by jons-aura** [comment]
- New project
- Console Application
- Target Framework .Net Core 5

![image](https://user-images.githubusercontent.com/35536253/111793944-227fab00-889c-11eb-9b22-de82bcad19c6.png)
![image](https://user-images.githubusercontent.com/35536253/111794164-51961c80-889c-11eb-974c-be86b42e1f5d.png)
![image](https://user-images.githubusercontent.com/35536253/111794619-cb2e0a80-889c-11eb-8709-3d9a5192a7eb.png)
Copied by CTRL-C  CTRL-V
![image](https://user-images.githubusercontent.com/35536253/111794694-e0a33480-889c-11eb-8b46-05332e53c964.png)
Renaming via F2
![image](https://user-images.githubusercontent.com/35536253/111794771-f1ec4100-889c-11eb-87c1-9d6c8eed027c.png)
Solution Explorer immediately after rename
![image](https://user-images.githubusercontent.com/35536253/111794835-0597a780-889d-11eb-8882-59c9440934c5.png)
.csproj contents
```
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="Class1 - Copy.cs" />
    <Compile Include="Class1.cs" />
  </ItemGroup>

</Project>
```

**Comment by drewnoakes** [comment]
@jons-aura thank you very much. That is perfect.

@ocallesp you were looking at rename operations recently. Does this look familiar?

**Comment by jjmew** [comment]
@drewnoakes were you able to repro this? We tried it during triage and we couldn't repro it

**Comment by drewnoakes** [comment]
I did not try to repro this.

Trying now in a newer build of VS I am not able to repro it.

@jons-aura are you able to upgrade to 16.9.x to see if the issue has been fixed?

**Comment by jons-aura** [comment]
@drewnoakes I can still reproduce it with 16.9.2

**Comment by drewnoakes** [comment]
@jons-aura do you have any extensions installed?

**Comment by jons-aura** [comment]
@drewnoakes I couldn't find a quick way to get them via text but here's a screenshot of the list
![image](https://user-images.githubusercontent.com/35536253/112317211-33f0fa80-8c82-11eb-9f6a-25af9f29509c.png)

**Comment by drewnoakes** [comment]
If you run `devenv.exe /SafeMode` it'll launch without extensions. I'd be curious if that fixes the problem.

**Comment by jons-aura** [comment]
@drewnoakes I can't open the solution in safe mode. The project shows as "needs migration" in Solution Explorer.

**Comment by drewnoakes** [comment]
@jons-aura thank you for bringing that to our attention. I've filed this as a bug here https://github.com/dotnet/project-system/issues/7046

In the meantime I'm perplexed as to what's causing this behaviour. Are you able to verify whether this occurs on another machine?

Does the error state eventually resolve itself, or does it remain with the red cross on the item indefinitely?

**Comment by jons-aura** [comment]
I would have sworn I replied to this already. I spent some time removing extensions and seeing if the problem kept happening - so far it is - but I asked a coworker to try and reproduce it and they couldn't. I'll try to find some time to poke at this some more but if I don't reply in a week or two feel free to close this since it doesn't seem to be a common issue and it's mostly just annoying.

**Comment by drewnoakes** [comment]
The fix for `/SafeMode` will be in 16.10 preview 2, so if you download that once it becomes available (soon I believe) then you could use that to exclude extensions from the picture.

Other things to look for on your machine:
- The presence of an `.editorconfig` file somewhere in the ancestry of your project
- The presence of `Directory.Build.props` or `Directory.Build.targets` files somewhere in the ancestry of your project

You could try copying your entire repository to another folder (e.g. `C:\foo`) then deleting all `bin`/`obj`/`.vs` folders. If you're using git, `git clean -xdf` can help.

**Comment by jons-aura** [comment]
I got a new system and reinstalled everything and can no longer reproduce this issue.
